### PR TITLE
Added polymer data and description.

### PR DESCRIPTION
**Pull request by AdrianM0**
Added the Bicerano dataset with multiple targets, both experimental and MD-calculated.

### CODE REVIEW

File: data/tabular/bicerano_dataset/HT_MD_polymer_properties.xlsx
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/HT_MD_polymer_properties.xlsx:1]: could you please use a csv or add the xlsx dependency for pandas in the requirements?

File: data/tabular/bicerano_dataset/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    refined optimized potentials for liquid simulation (OPLS) OPLS3e force field to
+    calculate glass transition temperatures (Tgs) of 315 polymers for which Bicerano 
+    reported experimental values.
+targets:
+    - id: Tg_exp
+      description: experimental glass transition T
+      units: K
+      type: float
+      names:
+          - noun: glass transition temperature - exp
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:14]: this is a text that is supposed to be part of the language model prompt, e.g. `what is the {noun} of {polymer}`. The `-exp` is a bit strange in this case.

File: data/tabular/bicerano_dataset/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      type: float
+      names:
+          - noun: glass transition temperature - exp 
+      uris:
+    - id: Tg_calc
+      description: calculated glass transition T
+      units: K
+      type: float
+      names:
+          - noun: glass transition temperature - calc
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:21]: same here

File: data/tabular/bicerano_dataset/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      uris:
+    - id: Tg_calc
+      description: calculated glass transition T
+      units: K
+      type: float
+      names:
+          - noun: glass transition temperature - calc 
+      uris:
+    - id: Tg_calc_std
+      description: calculated std glass transition T
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:24]:
```suggestion
    - id: Tg_calc_std
      description: calculated std glass transition T 
      sampled: False

```

File: data/tabular/bicerano_dataset/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      units: K
+      type: float
+      names:
+          - noun: glass transition temperature - calc 
+      uris:
+    - id: Tg_calc_std
+      description: calculated std glass transition T 
+      units: K
+      type: float
+      names:
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:27]:
```suggestion
      sampled: False
      names:

```

File: data/tabular/bicerano_dataset/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+          - noun: calculated std glass transition temperature 
+      uris:        
+    - id: rho_300K_exp
+      description: experiment density at 300K 
+      units: g/cc
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:32]:
```suggestion
      units: g/cm^3

```

File: data/tabular/bicerano_dataset/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+      type: float
+      names:
+          - noun: exp density 300K
+    - id: rho_300K_calc
+      description: calculated density at 300K 
+      units: g/cc
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:38]: same here

File: data/tabular/bicerano_dataset/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+      type: float
+      names:
+          - noun: calc density 300K
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:41]: think that it should be part of a prompt

File: data/tabular/bicerano_dataset/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    - id: glass_CTE_calc
+      description: calculated glass coef of thermal expansion (CTE) at 300K 
+      units: K
+      type: float
+      names:
+          - noun: calc density 300K
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:47]: this doesn't match the description

File: data/tabular/bicerano_dataset/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+    - id: glass_CTE_calc_std
+      description: std - calculated glass coef of thermal expansion (CTE) at 300K 
+      units: K
+      type: float
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:51]:
```suggestion
      type: float
     sampled: False
```

File: data/tabular/bicerano_dataset/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+      description: 
+      units: K
+      type: float
+      names:
+          - noun: calc density 300K
+    - id: rubber_CTE_calc_std
+      description: std - calculated glass coef of thermal expansion (CTE) at 300K 
+      units: K
+      type: float
+      names:
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:64]: same here

File: data/tabular/bicerano_dataset/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+          - noun: calc density 300K
+    - id: rubber_CTE_calc
+      description: 
+      units: K
+      type: float
+      names:
+          - noun: calc density 300K
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:59]: doesn't match description

File: data/tabular/bicerano_dataset/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+    - id: rubber_CTE_calc_std
+      description: std - calculated glass coef of thermal expansion (CTE) at 300K 
+      units: K
+      type: float
+      names:
+          - noun: calc density 300K
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:65]: doesn't match description

File: data/tabular/bicerano_dataset/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+benchmarks:
+    - name: 
+      link: 
+      split_column: 
+identifiers:
+    - id: SMILES
+      type: SMILES
+      description: SMILES
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:73]: is this really a plain monomer smiles?

File: data/tabular/bicerano_dataset/transform.py [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+                     "glass_CTE_calc", 
+                     "glass_CTE_calc_std", 
+                     "rubber_CTE_calc", 
+                     "rubber_CTE_calc_std"]
+    
+    clean_data.columns = clean_columns
+ 
+ 
+    clean_data["SMILES"] = clean_data["SMILES"].str.replace("[Ce]", "[*]", regex=False)
+    clean_data["SMILES"] = clean_data["SMILES"].str.replace("[Th]", "[*]", regex=False)
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/transform.py:29]: Ok, it is not a plain monomer SMILES. 

so I'd suggest two things: 

1. Add the `PSMILES` type in the data schema
2. Add the monomer SMILES (adding back leaving groups) 
3. Add templates that make clear that those are homopolymers (?) with this polymer SMILES and/or monomer SMILES 
4. You can add additional templates for other tasks such as conversational prompts "give me a PSMILES with density X and Tg Y"

File: data/tabular/bicerano_dataset/transform.py [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+    
+    clean_data.to_excel("HT_MD_Polymers_clean.xlsx")
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/transform.py:32]: in the end, we want a clean `csv` as the output (check some if the other `transform.py` scripts

File: data/tabular/bicerano_dataset/transform.py [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+
+transform_data()
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/transform.py:34]:
```suggestion
if __name__ == '__main__':
	transform_data()
```

**Review by kjappelbaum — CHANGES_REQUESTED** [review]
Nice job. I left a few comments that should clarify how the sampling pipeline works. The contribution guide is not always very clear.

File: data/tabular/bicerano_dataset/HT_MD_polymer_properties.xlsx
Comment (AdrianM0) [reply data/tabular/bicerano_dataset/HT_MD_polymer_properties.xlsx:1]: done, replaced xlsx with csv.

File: data/tabular/bicerano_dataset/meta.yaml [hunk 17]
```diff
@@ -0,0 +1,95 @@
...
+    - id: rubber_CTE_calc_std
+      description: computed standard deviation of rubber CTE at 300K 
+      sampled: False
+      units: K
+      type: float
+      names:
+          - computed standard deviation of rubber coefficient of thermal expansion (CTE) at 300K 
+identifiers:
+    - id: PSMILES
+      type: PSMILES
```
Comment (kjappelbaum) [inline data/tabular/bicerano_dataset/meta.yaml:69]: this has to be changed in the data schema, too https://github.com/OpenBioML/chemnlp/blob/0710cf53cbf72dd92e97577d65b453cbb118b3e0/src/chemnlp/data_val/model.py#L9

**Comment by kjappelbaum** [comment]
did you run the `pre-commit`?

**Comment by AdrianM0** [comment]
> Thanks for the changes! Now it is very, very close to ideal! (and I guess most things were due to unclear contribution guide)
> 
> Did you run the `pre-commit`?

did now

File: data/tabular/bicerano_dataset/meta.yaml (hunk 17 above)
Comment (AdrianM0) [reply data/tabular/bicerano_dataset/meta.yaml]: done

**Comment by kjappelbaum** [comment]
>Did you run the pre-commit? 

hmm.... the GitHub action did still complain, I'll try to check it out. If you used the the most recent versions then there should be no issue.

**Comment by kjappelbaum** [comment]
If you make PRs in the future, it will be cleaner if you make a new branch in your fork and then, from this branch, make a pull request

**Pull request merged**